plugins/node/opentelemetry-instrumentation-connect/src/instrumentation.ts

```diff
@@ -120,16 +120,16 @@ export class ConnectInstrumentation extends InstrumentationBase<Server> {
       if (!instrumentation.isEnabled()) {
         return (middleWare as any).apply(this, arguments);
       }
-      const [reqArgIdx, resArgIdx, nextArgIdx] = isErrorMiddleware
-        ? [1, 2, 3]
-        : [0, 1, 2];
-      const req = arguments[reqArgIdx] as IncomingMessage;
+      const [resArgIdx, nextArgIdx] = isErrorMiddleware ? [2, 3] : [1, 2];
       const res = arguments[resArgIdx] as ServerResponse;
       const next = arguments[nextArgIdx] as NextFunction;
 
       const rpcMetadata = getRPCMetadata(context.active());
       if (routeName && rpcMetadata?.type === RPCType.HTTP) {
-        rpcMetadata.span.updateName(`${req.method} ${routeName || '/'}`);
+        setRPCMetadata(
+          context.active(),
+          Object.assign(rpcMetadata, { route: routeName || '/' })
```

Comment:
I'm super new to connect so I'm not sure how this will behave for a nested route.

should we expand the RpcMetadata's route instead?

Maybe another PR?

Reply:
I think we should report the final route (e.g. `/foo/bar` in above case) but I would move this to a separate PR.